### ISSUE TITLE
broken goals from cmd-line

### ISSUE THREAD

**Issue by mvanini** [CycloneDX/cyclonedx-maven-plugin#180]
In last version 2.6.0 goal makeAggregateBom (but I suspect other goals too) is "broken" if you execute them directly from command line: while executing the goal of this plugin, it start the build from scratch and try to compile and do tests while this is not necessary! 

In our CI-System we use the plugin directly from command line, because we have the requisite of not edit the pom to including configuration of the plugin.

Now we have explicitated plugin version to 2.5.3 and all is back to work, but new version of the plugin fix a problem with SNAPSHOT identification (it now puts in the BOM the SNAPSHOT base-version declared in POMs, instead of the resolved artifact with timestamp-version) and we want to introduce in our pipelines

I paste here ours logs.

The old one with 2.5.3:
```
mvn -B -s /home/jenkins/agent/workspace/ci@tmp/config3030012475029251855tmp org.cyclonedx:cyclonedx-maven-plugin:2.5.3:makeAggregateBom -DschemaVersion=1.2 -DincludeTestScope=false -DSPDXParser.OnlyUseLocalLicenses=true
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------------< com.exaxmple:my-project >-----------------------
[INFO] Building My Project 0.0.25-SNAPSHOT
[INFO] --------------------------------[ war ]---------------------------------
[INFO] 
[INFO] --- cyclonedx-maven-plugin:2.5.3:makeAggregateBom (default-cli) @ my-project ---
[INFO] CycloneDX: Creating BOM
[INFO] CycloneDX: Writing BOM (XML): /home/jenkins/agent/workspace/ci/target/bom.xml
[INFO] CycloneDX: Validating BOM (XML): /home/jenkins/agent/workspace/ci/target/bom.xml
[INFO] CycloneDX: Writing BOM (JSON): /home/jenkins/agent/workspace/ci/target/bom.json
[INFO] CycloneDX: Validating BOM (JSON): /home/jenkins/agent/workspace/ci/target/bom.json
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  18.002 s
[INFO] Finished at: 2022-05-02T17:57:07+02:00
[INFO] ------------------------------------------------------------------------
```

And this is with 2.6.0 version:
```
mvn -B -s /home/jenkins/agent/workspace/ci@tmp/config4962994199716148515tmp org.cyclonedx:cyclonedx-maven-plugin:makeAggregateBom -DschemaVersion=1.2 -DincludeTestScope=false -DSPDXParser.OnlyUseLocalLicenses=true
 [INFO] Scanning for projects...
 [INFO] 
 [INFO] -----------------------< com.example:my-project >-----------------------
 [INFO] Building My Project 0.0.25-SNAPSHOT
 [INFO] --------------------------------[ war ]---------------------------------
 [INFO] 
 [INFO] >>> cyclonedx-maven-plugin:2.6.0:makeAggregateBom (default-cli) > test-compile @ my-project >>>
 [INFO] 
 [INFO] --- maven-enforcer-plugin:3.0.0-M3:enforce (enforce-versions) @ my-project ---
 [INFO] 
 [INFO] --- maven-enforcer-plugin:3.0.0-M3:enforce (enforce-dependencyConvergence) @ my-project ---
 [WARNING] 
 Dependency convergence error for org.hibernate:hibernate-core:5.4.33.Final paths to dependency are:
 +-com.example:my-project:0.0.25-SNAPSHOT
   +-org.hibernate:hibernate-core:5.4.33.Final
 and
 +-com.example:my-project:0.0.25-SNAPSHOT
   +-org.springframework.boot:spring-boot-starter-data-jpa:2.2.7.RELEASE
     +-org.hibernate:hibernate-core:5.4.15.Final
 
 [WARNING] Rule 0: org.apache.maven.plugins.enforcer.DependencyConvergence failed with message:
 Failed while enforcing releasability. See above detailed error message.
 [INFO] 
 [INFO] --- build-helper-maven-plugin:3.2.0:timestamp-property (timestamp-property) @ my-project ---
 [WARNING] Using platform locale (en_US actually) to format date/time, i.e. build is platform dependent!
 [INFO] 
 [INFO] --- maven-resources-plugin:3.1.0:copy-resources (default-resources) @ my-project ---
 [INFO] Using 'UTF-8' encoding to copy filtered resources.
 [INFO] Copying 6 resources
 [INFO] Copying 46 resources
 [INFO] 
 [INFO] --- maven-resources-plugin:3.1.0:resources (default-resources) @ my-project ---
 [INFO] Using 'UTF-8' encoding to copy filtered resources.
 [INFO] Copying 6 resources
 [INFO] Copying 46 resources
 [INFO] 
 [INFO] --- jacoco-maven-plugin:0.8.5:prepare-agent (pre-unit-tests) @ my-project ---
 [INFO] argLine set to -javaagent:/home/jenkins/agent/workspace/ci/.m2/org/jacoco/org.jacoco.agent/0.8.5/org.jacoco.agent-0.8.5-runtime.jar=destfile=/home/jenkins/agent/workspace/ci/target/jacoco/test/test.exec -Djava.security.egd=file:/dev/./urandom -Xmx256m
 [INFO] 
 [INFO] --- properties-maven-plugin:1.0.0:read-project-properties (default) @ my-project ---
 [INFO] 
 [INFO] --- build-helper-maven-plugin:3.2.0:parse-version (parse-version) @ my-project ---
 [INFO] 
 [INFO] --- git-commit-id-plugin:4.0.0:revision (default) @ my-project ---
 [INFO] 
 [INFO] --- maven-apt-plugin:1.0.4:process (process-dsl) @ my-project ---
 [INFO] 
 [INFO] --- build-helper-maven-plugin:3.2.0:add-source (add-sources) @ my-project ---
 [INFO] Source directory: /home/jenkins/agent/workspace/ci/target/generated-sources/annotations added.
 [INFO] 
 [INFO] --- spring-boot-maven-plugin:2.2.7.RELEASE:build-info (default) @ my-project ---
 [INFO] 
 [INFO] --- exec-maven-plugin:1.5.0:exec (run npm install) @ my-project ---
 [ERROR] Command execution failed.
 java.io.IOException: Cannot run program "npm" (in directory "/home/jenkins/agent/workspace/ci"): error=2, No such file or directory
     at java.lang.ProcessBuilder.start (ProcessBuilder.java:1128)
     at java.lang.ProcessBuilder.start (ProcessBuilder.java:1071)
     at java.lang.Runtime.exec (Runtime.java:592)
     at org.apache.commons.exec.launcher.Java13CommandLauncher.exec (Java13CommandLauncher.java:61)
     at org.apache.commons.exec.DefaultExecutor.launch (DefaultExecutor.java:279)
     at org.apache.commons.exec.DefaultExecutor.executeInternal (DefaultExecutor.java:336)
     at org.apache.commons.exec.DefaultExecutor.execute (DefaultExecutor.java:166)
     at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:764)
     at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:711)
     at org.codehaus.mojo.exec.ExecMojo.execute (ExecMojo.java:289)
     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
     at org.apache.maven.lifecycle.internal.MojoExecutor.executeForkedExecutions (MojoExecutor.java:355)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:200)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
     at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
     at java.lang.reflect.Method.invoke (Method.java:566)
     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
 Caused by: java.io.IOException: error=2, No such file or directory
     at java.lang.ProcessImpl.forkAndExec (Native Method)
     at java.lang.ProcessImpl.<init> (ProcessImpl.java:340)
     at java.lang.ProcessImpl.start (ProcessImpl.java:271)
     at java.lang.ProcessBuilder.start (ProcessBuilder.java:1107)
     at java.lang.ProcessBuilder.start (ProcessBuilder.java:1071)
     at java.lang.Runtime.exec (Runtime.java:592)
     at org.apache.commons.exec.launcher.Java13CommandLauncher.exec (Java13CommandLauncher.java:61)
     at org.apache.commons.exec.DefaultExecutor.launch (DefaultExecutor.java:279)
     at org.apache.commons.exec.DefaultExecutor.executeInternal (DefaultExecutor.java:336)
     at org.apache.commons.exec.DefaultExecutor.execute (DefaultExecutor.java:166)
     at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:764)
     at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:711)
     at org.codehaus.mojo.exec.ExecMojo.execute (ExecMojo.java:289)
     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
     at org.apache.maven.lifecycle.internal.MojoExecutor.executeForkedExecutions (MojoExecutor.java:355)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:200)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
     at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
     at java.lang.reflect.Method.invoke (Method.java:566)
     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
 [INFO] ------------------------------------------------------------------------
 [INFO] BUILD FAILURE
 [INFO] ------------------------------------------------------------------------
 [INFO] Total time:  7.784 s
 [INFO] Finished at: 2022-05-02T09:20:58+02:00
 [INFO] ------------------------------------------------------------------------
 [ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.5.0:exec (run npm install) on project my-project: Command execution failed. Cannot run program "npm" (in directory "/home/jenkins/agent/workspace/ci"): error=2, No such file or directory -> [Help 1]
 [ERROR] 
 [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
 [ERROR] Re-run Maven using the -X switch to enable full debug logging.
 [ERROR] 
 [ERROR] For more information about the errors and possible solutions, please read the following articles:
 [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```
It is the same project, run in the same machine (a container with same image) and at the same git revision.

I know that the build is broken for a missing requirement of the system (npm in thiscase) but what is wrong is build the project while I'm not requesting it! 

I think that #153 was the cause of regression and moreover we explicitly use the parameter for not including test-scoped dependency!

**Comment by stevespringett** [comment]
Thanks for reporting. This should be fixed in 2.6.1.

**Comment by stevespringett** [comment]
BTW, `-DSPDXParser.OnlyUseLocalLicenses` is a legacy parameter and no longer necessary.
